File: hackerrank/30-days-of-code/day-26.py
# -*- coding: utf-8 -*-
# Objective
# Today's challenge puts your understanding of nested conditional statements to the test. You already have the knowledge to complete this challenge, but check out the Tutorial tab for a video on testing!

# Task
# Your local library needs your help! Given the expected and actual return dates for a library book, create a program that calculates the fine (if any). The fee structure is as follows:

# If the book is returned on or before the expected return date, no fine will be charged (i.e.: fine=0)fine=0).
# If the book is returned after the expected return day but still within the same calendar month and year as the expected return date, fine=15 Hackos × (the number of days late)fine=15 Hackos × (the number of days late).
# If the book is returned after the expected return month but still within the same calendar year as the expected return date, the fine=500 Hackos × (the number of months late)fine=500 Hackos × (the number of months late).
# If the book is returned after the calendar year in which it was expected, there is a fixed fine of 10000 Hackos10000 Hackos.
# Input Format

# The first line contains 33 space-separated integers denoting the respective dayday, monthmonth, and yearyear on which the book was actually returned.
# The second line contains 33 space-separated integers denoting the respective dayday, monthmonth, and yearyear on which the book was expected to be returned (due date).

# Constraints

# 1≤D≤311≤D≤31
# 1≤M≤121≤M≤12
# 1≤Y≤30001≤Y≤3000
# It is guaranteed that the dates will be valid Gregorian calendar dates.It is guaranteed that the dates will be valid Gregorian calendar dates.
# Output Format

# Print a single integer denoting the library fine for the book received as input.

# Sample Input

# 9 6 2015
# 6 6 2015
# Sample Output

# 45
# Explanation

# Given the following return dates:
# Actual: Da=9,Ma=6,Ya=2015Da=9,Ma=6,Ya=2015
# Expected: De=6,Me=6,Ye=2015De=6,Me=6,Ye=2015
# Because Ye≡YaYe≡Ya, we know it is less than a year late.
# Because Me≡MaMe≡Ma, we know it's less than a month late.
# Because De<DaDe<Da, we know that it was returned late (but still within the same month and year).

# Per the library's fee structure, we know that our fine will be 15 Hackos × (# days late)15 Hackos × (# days late). We then print the result of 15×(Da−De)=15×(9−6)=4515×(Da−De)=15×(9−6)=45 as our output.

File: hackerrank/30-days-of-code/day-26.java
// To calculate the fine, we must first compare the  and the , then the  to the , and finally the  to the . The should be initialized to .

// Years:
// If , then the book was returned early and .
// If , then .
// If , then the book was either returned on time or was returned less than a year late; to determine this, we move on to our Months logic.

// Months:
// If , then the book was returned early and .
// If , then the book was returned some number of months late and .
// If , then the book was either returned on time or less than a month late; to determine this, we move on to our Days logic.

// Days:
// If , then .
// If , then .

import java.util.*;

class BookReturnDate{
    public int day;
    public int month;
    public int year;

    BookReturnDate(int day, int month, int year){
        this.day = day;
        this.month = month;
        this.year = year;
    }
}

public class Solution {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        BookReturnDate actual = new BookReturnDate(scan.nextInt(), scan.nextInt(),scan.nextInt());
        BookReturnDate expected = new BookReturnDate(scan.nextInt(), scan.nextInt(),scan.nextInt());
        scan.close();

        int fine = 0;

        // Returned within the current year:
        if(expected.year == actual.year){
            // Returned 1+ months late in current year
            if(expected.month < actual.month){
                fine = (actual.month - expected.month) * 500;
            }
            // Returned 1+ days late within the current month
            else if( (expected.month == actual.month)
                    && (expected.day < actual.day) ){
                fine = (actual.day - expected.day) * 15;
            }
            // Else it is implied that the book was returned early
        }
        else if(expected.year < actual.year){
            // Returned 1+ years late
            fine = 10000;
        }
        // Else it is implied that the book was returned 1+ years early

        System.out.println(fine);

    }
}

// Problem Tester's code :
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args)throws IOException {
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        String s1=br.readLine().trim();
        String s2=br.readLine().trim();
        StringTokenizer st=new StringTokenizer(s1);
        int d1=Integer.parseInt(st.nextToken());
        int m1=Integer.parseInt(st.nextToken());
        int y1=Integer.parseInt(st.nextToken());
        st=new StringTokenizer(s2);
        int d2=Integer.parseInt(st.nextToken());
        int m2=Integer.parseInt(st.nextToken());
        int y2=Integer.parseInt(st.nextToken());
        int fine=0;
        if(y1>y2){
            fine=10000;
        }
        else if(m1>m2){
            fine=(m1-m2)*500;
        }
        else if(d1>d2){
            fine=(d1-d2)*15;
        }
        System.out.println(fine);



    }
}